pkg/api/datavolume/store.go

```diff
@@ -0,0 +1,44 @@
+package datavolume
+
+import (
+	"fmt"
+
+	"github.com/rancher/apiserver/pkg/apierror"
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/wrangler/pkg/schemas/validation"
+
+	cdiv1beta1 "github.com/rancher/harvester/pkg/generated/controllers/cdi.kubevirt.io/v1beta1"
+)
+
+const (
+	VMKindName = "VirtualMachine"
+)
+
+type dvStore struct {
+	types.Store
+	dvCache cdiv1beta1.DataVolumeCache
+}
+
+func (s *dvStore) Delete(request *types.APIRequest, schema *types.APISchema, id string) (types.APIObject, error) {
+	if err := s.delete(request.Namespace, request.Name); err != nil {
+		return types.APIObject{}, apierror.NewAPIError(validation.ServerError, err.Error())
+	}
+	return s.Store.Delete(request, request.Schema, id)
+}
+
+func (s *dvStore) delete(namespace, name string) error {
```

Comment:
we don't do delete in this function, so use a proper name for it. For example, canDelete().